public/app/features/search/page/PanelSearchPage.tsx

```diff
@@ -0,0 +1,225 @@
+    const searcher = getGrafanaSearcher();
+    const panelResults = await searcher.search({
+      query,
+      kind: ['panel'],
+      limit: 12,
+    });
+
+    const panels: PanelModel[] = [];
+    for (let i = 0; i < panelResults.view.length; i++) {
+      const p = panelResults.view.get(i);
+      const panel = await this.findPanel(p);
+      panels.push(panel);
+    }
+    return {
+      totalRows: panelResults.totalRows,
+      panels,
+      scene: panelsToScene(panels),
+    };
+  }
+
+  async getDashboard(uid: string): Promise<DashboardDTO> {
+    let dash = this.dashboards.get(uid);
+    if (dash) {
+      return Promise.resolve(dash);
+    }
+    dash = await backendSrv.getDashboardByUid(uid);
+    dash.meta.canEdit = false; // avoids popup
+    dash.meta.canSave = false; // avoids popup
+    dash.meta.fromScript = true; // avoids popup
+    this.dashboards.set(uid, dash);
+    return dash;
+  }
+
+  async findPanel(p: DashboardQueryResult): Promise<PanelModel> {
+    if (!config.panels[p.panel_type]) {
+      return this.getErrorPanel('Unknown panel type ' + p.panel_type, p);
+    }
+
+    const idx = p.uid.indexOf('#');
+    if (idx < 2) {
+      return this.getErrorPanel('expected # in the UID', p);
+    }
+
+    const dashuid = p.uid.substring(0, idx);
+    const panelid = parseInt(p.uid.substring(idx + 1), 10);
+    const dash = await this.getDashboard(dashuid);
+    const found = dash?.dashboard?.panels?.find((p) => p.id === panelid);
+    if (!found) {
+      return this.getErrorPanel('Unable to find: ' + p.uid, p);
+    }
+
+    const links = [
+      {
+        title: 'Dashboard',
+        url: p.url,
+      },
+    ];
+
+    if (found.links && Array.isArray(found.links)) {
+      for (const v of found.links) {
+        links.push(v);
+      }
+    }
+
+    return {
+      ...found,
+      id: this.counter++,
+      key: p.uid,
+      links,
+    };
+  }
+
+  getErrorPanel(err: string, p: DashboardQueryResult): PanelModel {
+    const v = {
+      id: this.counter++,
+      key: p.uid,
+      title: `ERROR: ${p.name}`,
+      type: 'text',
+      options: {
+        content: `ERROR:  ${err}`,
+        mode: TextMode.Markdown,
+      },
+      fieldConfig: {
+        defaults: {},
+        overrides: [],
+      },
+    };
+    return v as unknown as PanelModel;
+  }
+}
+
+export function panelsToScene(results: PanelModel[]): Scene {
+  let x = 0;
+  let panelW = 24 / 3; //8
+
+  const state = {
+    title: 'Search results',
+    layout: new SceneGridLayout({
+      children: results.map((panel, idx) => {
+        if (idx > 0) {
+          x += panelW;
+        }
+        if (x > 20) {
+          x = 0;
+        }
+        return new VizPanel({
+          isResizable: false,
+          isDraggable: false,
+          size: {
+            x,
+            y: 0,
+            width: panelW,
+            height: 8, // grid pos
+            minHeight: '200px',
+          },
```

Comment:
After https://github.com/grafana/grafana/pull/60437 I am going to add a wrap option to SceneFlexLayout then  I think that is a better (and easier) option than the Grid layout